db/migrate_core_cards/20141230204340_uri_codename.rb

```diff
@@ -0,0 +1,11 @@
+# -*- encoding : utf-8 -*-
+
+class UriCodename < Wagn::Migration
+  def up
+    contentedly do
+      puts "adding new codename card for URI"
+      codename = 'URI'
+      Card.create! :type_id=>Card::CardtypeID, :name=>codename, :codename=>codename.to_name.key
```

Comment:
I like the idea of handling this in a standard way; not sure it needs to be migration specific.  Maybe something like :autoname=>true ?

right, codename collisions aren't really a worry here, imo.

btw, this code is a little harder to read than it needs to be. the variable "codename" is used for the name version.  not a big deal, but since we're here...